permissions/resource_permissions.go

```diff
@@ -215,6 +215,16 @@ func (a PermissionsAPI) Delete(objectID string) error {
 				accl.AccessControlList = append(accl.AccessControlList, change)
 			}
 		}
+
+		// handle special case when we add extra permission to a user home dir
+		if v, ok := d.GetOk("directory_path"); ok {
```

Comment:
We may need to disallow changes only for the current user - then we'll need to add a check like `if me == accessControl.UserName || me == accessControl.ServicePrincipalName {` like [below](https://github.com/databricks/terraform-provider-databricks/pull/3586/files#diff-e21ed6630273de6b0cb95643ddfc8cda8082f1ed5c5ce39b53210622ad90f55aR339). We also need to handle the case of home directory of the service principal, not only ordinary users.